third_party/accessibility/ax/platform/ax_platform_node_textprovider_win_unittest.cc

```diff
@@ -0,0 +1,946 @@
+                  button, button_img, button_text,  dialog_detail};
+
+  Init(update);
+
+  AXNode* root_node = GetRootAsAXNode();
+  AXNode* dialog_node = root_node->children()[0];
+  AXPlatformNodeWin* owner =
+      static_cast<AXPlatformNodeWin*>(AXPlatformNodeFromNode(root_node));
+  BASE_DCHECK(owner);
+
+  ComPtr<IRawElementProviderSimple> root_node_raw =
+      QueryInterfaceFromNode<IRawElementProviderSimple>(root_node);
+  ComPtr<IRawElementProviderSimple> dialog_node_raw =
+      QueryInterfaceFromNode<IRawElementProviderSimple>(dialog_node);
+
+  // Call RangeFromChild on the root with the dialog child passed in.
+  ComPtr<ITextProvider> text_provider;
+  EXPECT_HRESULT_SUCCEEDED(
+      root_node_raw->GetPatternProvider(UIA_TextPatternId, &text_provider));
+
+  ComPtr<ITextRangeProvider> text_range_provider;
+  EXPECT_HRESULT_SUCCEEDED(text_provider->RangeFromChild(dialog_node_raw.Get(),
+                                                         &text_range_provider));
+  SetOwner(owner, text_range_provider.Get());
+
+  base::win::ScopedBstr text_content;
+  EXPECT_HRESULT_SUCCEEDED(
+      text_range_provider->GetText(-1, text_content.Receive()));
+  EXPECT_EQ(base::WideToUTF16(text_content.Get()),
+            u"Dialog label.Dialog description." + kEmbeddedCharacterAsString +
+                u"ok.Some more detail " + u"about dialog.");
+
+  // Check the reverse relationship that GetEnclosingElement on the text range
+  // gives back the dialog.
+  ComPtr<IRawElementProviderSimple> enclosing_element;
+  EXPECT_HRESULT_SUCCEEDED(
+      text_range_provider->GetEnclosingElement(&enclosing_element));
+  EXPECT_EQ(enclosing_element.Get(), dialog_node_raw.Get());
+}
+
+TEST_F(AXPlatformNodeTextProviderTest, NearestTextIndexToPoint) {
+  AXNodeData text_data;
+  text_data.id = 2;
+  text_data.role = ax::mojom::Role::kInlineTextBox;
+  text_data.SetName("text");
+  // spacing: "t-e-x---t-"
+  text_data.AddIntListAttribute(ax::mojom::IntListAttribute::kCharacterOffsets,
+                                {2, 4, 8, 10});
+
+  AXNodeData root_data;
+  root_data.id = 1;
+  root_data.role = ax::mojom::Role::kRootWebArea;
+  root_data.relative_bounds.bounds = gfx::RectF(1, 1, 2, 2);
+  root_data.child_ids.push_back(2);
+
+  Init(root_data, text_data);
+
+  AXNode* root_node = GetRootAsAXNode();
+  AXNode* text_node = root_node->children()[0];
+
+  struct NearestTextIndexTestData {
+    AXNode* node;
+    struct point_offset_expected_index_pair {
+      int point_offset_x;
+      int expected_index;
+    };
+    std::vector<point_offset_expected_index_pair> test_data;
+  };
+  NearestTextIndexTestData nodes[] = {
+      {text_node,
+       {{0, 0}, {2, 0}, {3, 1}, {4, 1}, {5, 2}, {8, 2}, {9, 3}, {10, 3}}},
+      {root_node,
+       {{0, 0}, {2, 0}, {3, 0}, {4, 0}, {5, 0}, {8, 0}, {9, 0}, {10, 0}}}};
+  for (auto data : nodes) {
+    if (!data.node->IsText() && !data.node->data().IsTextField()) {
+      continue;
+    }
```

Comment:
Is this necessary? If looks like the original test passed without this

Reply:
I think it is because we treat the `kRootWebArea` role differently than in Chrome.